src/main/java/randoop/util/Randomness.java

```diff
@@ -90,6 +92,51 @@ public static int nextRandomInt(int i) {
     throw new BugInRandoopException();
   }
 
+  public static <T extends WeightedElement> T randomMemberWeighted(
+      SimpleList<T> list, Map<WeightedElement, Double> weights) {
+
+    // Find interval length.
+    double max = 0;
+    List<Double> cumulativeWeights = new ArrayList<>();
+    cumulativeWeights.add(0.0);
+    for (int i = 0; i < list.size(); i++) {
+      Double weight = weights.get(list.get(i));
+      if (weight == null) {
+        Log.logLine("randoop.util.Randomness: weight was null");
+        weight = list.get(i).getWeight();
+      }
+      if (weight <= 0) throw new BugInRandoopException("weight was " + weight);
+      cumulativeWeights.add(cumulativeWeights.get(cumulativeWeights.size() - 1) + weight);
+      max += weight;
+    }
+    assert max > 0;
+
+    // Select a random point in interval and find its corresponding element.
+    totalCallsToRandom++;
+    if (Log.isLoggingOn()) {
+      Log.logLine("randoop.util.Randomness: " + totalCallsToRandom + " calls so far.");
+    }
+    double randomPoint = Randomness.random.nextDouble() * max;
+
+    return list.get(binarySearchForIndex(list, cumulativeWeights, randomPoint));
+  }
+
+  public static int binarySearchForIndex(
```

Comment:
What is the relationship between the two routines named `binarySearchForIndex` that you added?  Their source code is identical, which is a code smell.  I suggest that you abstract them into a single routine, to avoid code duplication.